src/commands/templates/delete.ts

```diff
@@ -0,0 +1,225 @@
+/* eslint-disable @typescript-eslint/member-delimiter-style */
+
+import { confirm, input, select } from "@inquirer/prompts";
+import ora from "ora";
+import { ServerClient } from "postmark";
+import { TemplateTypes } from "postmark/dist/client/models";
+
+import {
+  fatalError,
+  log,
+  logError,
+  pluralize,
+  validateToken,
+} from "../../utils";
+
+interface TemplateDeleteArguments {
+  serverToken: string;
+  requestHost: string;
+}
+
+export interface TemplateListOptions {
+  requestHost: string;
+  sourceServer: string;
+  templateType: TemplateTypes | undefined;
+}
+
+export async function handler(args: TemplateDeleteArguments): Promise<void> {
+  const serverToken = await validateToken(args.serverToken);
+  const requestHost = args.requestHost;
+  deletePrompts(serverToken, requestHost);
+}
+
+/**
+ * Ask user a series of questions before deleting templates.
+ */
+
+async function deletePrompts(
+  serverToken: string,
+  requestHost: string
+): Promise<void> {
+  const choice = await select({
+    message: "Choose how you want to delete templates:",
+    choices: [
+      {
+        name: "Delete templates by id or alias",
+        description:
+          "NOTE: mixing ids and aliases on the same request is NOT supported",
+        value: "byIdOrAlias",
+      },
+      {
+        name: "Delete all templates",
+        value: "all",
+      },
+    ],
+  });
+
+  /**
+   * if the user has selected to delete templates by id
+   */
+
+  if (choice === "byIdOrAlias") {
+    // TODO we probably want an actual validation
+    // against ids/aliases that exist in the server
+    const validateUserInput = async (input: string) => {
+      // do not allow empty strings
+      if (input) return true;
+      return false;
+    };
+
+    const userInput = await input({
+      message:
+        "Enter template id(s)/alias(es) - separate with commas if multiple:",
+      validate: validateUserInput,
```

Comment:
Maybe we could inline the validation if it's so simple?

```suggestion
      validate: (s) => s.trim().length > 0, // don't allow empty input
```